packages/flutter_tools/test/integration.shard/native_assets_test.dart

```diff
@@ -394,9 +396,35 @@ Future<Directory> createTestProject(String packageName, Directory tempDirectory)
   expect(packageDirectory.childDirectory('macos/'), isNot(exists));
   expect(packageDirectory.childDirectory('windows/'), isNot(exists));
 
+  await pinDependencies(packageDirectory.childFile('pubspec.yaml'));
+  await pinDependencies(
+      packageDirectory.childDirectory('example').childFile('pubspec.yaml'));
+
+  final ProcessResult result2 = await processManager.run(
+    <String>[
+      flutterBin,
+      'pub',
+      'get',
+    ],
+    workingDirectory: packageDirectory.path,
+  );
+  if (result2.exitCode != 0) {
+    throw Exception(
+      'flutter create failed: ${result2.exitCode}\n${result2.stderr}\n${result2.stdout}',
+    );
+  }
```

Comment:
we did this so often we created a custom matcher for it:

```suggestion
  expect(result2, const ProcessResultMatcher());
```